cmds/core/netcat/connect_test.go

```diff
@@ -20,6 +22,110 @@ import (
 	"github.com/u-root/u-root/pkg/netcat"
 )
 
+func TestConnectMode(t *testing.T) {
+	connectErr := errors.New("failed to connect")
+
+	tests := []struct {
+		name          string
+		network       string
+		address       string
+		connectFunc   func(output io.Writer, network, address string) error
+		expectedError error
```

Comment:
one little note here: go style tends to be brief, it's more comment to name this err and not expectedError.
See what you think. I personally prefer just err.